projects/batfish-common-protocol/src/main/java/org/batfish/datamodel/PrefixIp6Space.java

```diff
@@ -0,0 +1,61 @@
+package org.batfish.datamodel;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.base.MoreObjects;
+import com.google.common.cache.CacheBuilder;
+import com.google.common.cache.CacheLoader;
+import com.google.common.cache.LoadingCache;
+import javax.annotation.ParametersAreNonnullByDefault;
+import org.batfish.datamodel.visitors.GenericIp6SpaceVisitor;
+
+@ParametersAreNonnullByDefault
+public class PrefixIp6Space extends Ip6Space {
+  private static final LoadingCache<Prefix6, PrefixIp6Space> CACHE =
+      CacheBuilder.newBuilder()
+          .softValues()
+          .maximumSize(1 << 20)
+          .build(CacheLoader.from(PrefixIp6Space::new));
+
+  private static final String PROP_PREFIX = "prefix";
+  private final Prefix6 _prefix6;
+
+  @JsonCreator
+  static PrefixIp6Space create(@JsonProperty(PROP_PREFIX) Prefix6 prefix6) {
```

Comment:
> This needs to be private, and is usually called something like jacksonCreator to prevent it from being confused with create that should be called by any user written code.

Other class and tests cannot call it but is `private` correct.
Ipv4 version constructor is a private but `create` static method is public.